daemon/api_apps.go

```diff
@@ -45,8 +45,8 @@ var (
 		Path:        "/v2/apps",
 		GET:         getAppsInfo,
 		POST:        postApps,
-		ReadAccess:  openAccess{},
-		WriteAccess: authenticatedAccess{Polkit: polkitActionManage},
+		ReadAccess:  interfaceOpenAccess{Interfaces: []string{"snap-apps-control"}},
+		WriteAccess: interfaceAuthenticatedAccess{Interfaces: []string{"snap-apps-control"}, Polkit: polkitActionManage},
```

Comment:
I think this will need more discussion about the scope of the interface. It's not clear to me how it's intended to be used at this point. It basically gives an open access to manage apps from any snap plugging the interface. One could end up in a situation when a random ROS app can shut down your lxd instance by accident or intentionally.

Should perhaps the scope of snaps the request can act on be limited to a subset of all snaps in the system? Maybe there should be some match, .e.g the same publisher of the snap making the request and the snap that would be affected by it?

Reply:
I think the interface needs permission to use and after that you can still shoot yourself in the foot but that's IMO expected at this level.